src/signax/signatures.py

```diff
@@ -142,16 +208,28 @@ def scan_fn(last_sig, current_stream_signature):
         return signature_combine(bulk_signature, remainder_signature)
 
     # no remainder, just return the bulk
+    if flatten:
+        bulk_signature = utils.flatten(bulk_signature)
     return bulk_signature
 
 
+@partial(jax.jit, static_argnames=["depth", "stream", "flatten", "num_chunks"])
 def logsignature(
-    path: Float[Array, "path_len dim"], depth: int, stream: bool = False
+    path: Float[Array, "path_len dim"],
```

Comment:
I guess you want to make `logsignature` to handle the input `(batch_size, path_len, dim)` as well. 

In this case, we also need to handle the shape and need an extra `vmap`

Reply:
this should work automatically by dispatching to those checks in `signature`, which should do the right things I think! But let me know if there's logic I'm missing here.